File: bindings/lua/Polycode/CoreServices.lua
require "Polycode/EventDispatcher"

class "CoreServices" (EventDispatcher)


function CoreServices:createInstance()
	local retVal =  Polycode.CoreServices_createInstance(self.__ptr)
end

function CoreServices:Update(elapsed)
	local retVal = Polycode.CoreServices_Update(self.__ptr, elapsed)
end

function CoreServices:fixedUpdate()
	local retVal =  Polycode.CoreServices_fixedUpdate(self.__ptr)
end

function CoreServices:__delete()
	if self then Polycode.delete_CoreServices(self.__ptr) end
end

File: bindings/javascript/Polycode/CoreServices.js
require('Polycode/EventDispatcher')

function CoreServices() {
}


CoreServices.prototype = Object.create(EventDispatcher.prototype)

Duktape.fin(CoreServices.prototype, function (x) {
	if (x === CoreServices.prototype) {
		return;
	}
	Polycode.CoreServices__delete(x.__ptr)
})

CoreServices.prototype.createInstance = function() {
	Polycode.CoreServices_createInstance()
}

CoreServices.prototype.Update = function(elapsed) {
	Polycode.CoreServices_Update(this.__ptr, elapsed)
}

CoreServices.prototype.fixedUpdate = function() {
	Polycode.CoreServices_fixedUpdate(this.__ptr)
}
